crates/ethereum/src/utils.rs

```diff
@@ -0,0 +1,21 @@
+use pathfinder_common::prelude::*;
+use pathfinder_crypto::Felt;
+
+/// Converts a `Signed<256, 4>` integer to a `BlockNumber`
+pub(crate) fn get_block_number(block_number: alloy::primitives::Signed<256, 4>) -> BlockNumber {
+    let block_number = block_number.as_i64();
```

Comment:
`.as_i64()` panics if the number is outside the range representable by `i64`. Can't we use `.try_from()` instead to do a fallible conversion to u64? That way we could just return a `Result<BlockNumber, _>` and handle the invalid input data properly in the caller.